src/parser/parse-2-typedefs.cpp

```diff
@@ -34,7 +34,7 @@ Result<> parseTypeDefs(
   if (auto* err = built.getError()) {
     std::stringstream msg;
     msg << "invalid type: " << err->reason;
-    return ctx.in.err(decls.typeDefs[err->index].pos, msg.str());
+    return ctx.in.err(decls.subtypeDefs[err->index].pos, msg.str());
```

Comment:
Why is this correct? I'm not sure how I would have known to think this was wrong. We are in `parseTypeDefs`, so shouldn't the `typeDefs` be relevant..? We even loop over `typeDefs on line 29.

Reply:
Each entry in `typeDefs` corresponds to a single recursion group (which may consist of a single top-level type definition without `(rec...)`, whereas each entry in `subtypeDefs` corresponds to a single type definition, which may or may not actually contain `(sub ...)`.

Better names would perhaps be `recTypeDefs` and `typeDefs` rather than `typeDefs` and `subtypeDefs`. Do you think it's worth updating those names here or in a separate PR?

Reply:
I see, thanks. Either way sgtm. lgtm as is if you prefer separately.